pages/aoc.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useEffect } from 'react';
+import '@/app/output.css';
+
+// 7pm MST = 9pm EST = 2am UTC next day
+// Day 1 unlocks Dec 2 at 7pm MST (Dec 3 at 2am UTC)
+// Day 12 unlocks Dec 13 at 7pm MST (Dec 14 at 2am UTC)
+const YEAR = 2025;
+const DAYS = 12;
+const UNLOCK_HOUR_UTC = 2; // 2am UTC = 7pm MST = 9pm EST
+
+function getUnlockTime(day: number): Date {
+  // Day 1 unlocks on Dec 2, Day 2 on Dec 3, etc.
+  // At 2am UTC (which is 7pm MST the previous calendar day)
+  return new Date(Date.UTC(YEAR, 11, day + 2, UNLOCK_HOUR_UTC, 0, 0));
+}
+
+function isDayUnlocked(day: number, now: Date): boolean {
+  return now >= getUnlockTime(day);
+}
+
+function getNextUnlockDay(now: Date): number | null {
+  for (let day = 1; day <= DAYS; day++) {
+    if (!isDayUnlocked(day, now)) {
+      return day;
+    }
+  }
+  return null; // All days unlocked
+}
+
+function formatTimeRemaining(ms: number): string {
+  if (ms <= 0) return '00:00:00';
+
+  const seconds = Math.floor((ms / 1000) % 60);
+  const minutes = Math.floor((ms / (1000 * 60)) % 60);
+  const hours = Math.floor((ms / (1000 * 60 * 60)) % 24);
+  const days = Math.floor(ms / (1000 * 60 * 60 * 24));
+
+  if (days > 0) {
+    return `${days}d ${hours.toString().padStart(2, '0')}:${minutes.toString().padStart(2, '0')}:${seconds.toString().padStart(2, '0')}`;
+  }
+  return `${hours.toString().padStart(2, '0')}:${minutes.toString().padStart(2, '0')}:${seconds.toString().padStart(2, '0')}`;
+}
+
+const AOCTimer = () => {
+  const [now, setNow] = useState(new Date());
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setNow(new Date());
+    }, 1000);
+    return () => clearInterval(interval);
+  }, []);
+
+  const nextUnlockDay = getNextUnlockDay(now);
+  const nextUnlockTime = nextUnlockDay ? getUnlockTime(nextUnlockDay) : null;
+  const timeRemaining = nextUnlockTime ? nextUnlockTime.getTime() - now.getTime() : 0;
+
```

Comment:
You can format the next date and that could have help to notice the problem of the date above
```suggestion
  const nextDate = nextUnlockTime ? nextUnlockTime.toLocaleString('en-US', { day: 'numeric', month: 'short', year: 'numeric' }) : "";


```